### PR TITLE
Avoid using EM_IGNORE_SANITY in test code. NFC.

### PR DESCRIPTION
**Pull request by sbc100**
This isn't needed provided:
a. the fake tools were good enough to fool our checks
b. we clear the sanity file explicitly between each test iteration.

### CODE REVIEW

**Comment by sbc100** [comment]
View diff with ignore whitespace

File: tests/test_sanity.py [hunk 1]
```diff
@@ -308,31 +298,31 @@ def test_node(self):
...
-        f = open(self.in_dir('fake', 'nodejs'), 'w')
-        f.write('#!/bin/sh\n')
-        f.write('''if [ $1 = "--version" ]; then
+    for version, succeed in [('v0.8.0', False),
+                             ('v4.1.0', False),
+                             ('v4.1.1', True),
+                             ('v4.2.3-pre', True),
+                             ('cheez', False)]:
+      print(version, succeed)
+      try_delete(SANITY_FILE)
```
Comment (kripken) [inline tests/test_sanity.py:307]: was this for your own debugging? if not why is it needed now?

Comment (sbc100) [reply tests/test_sanity.py:307]: You mean the `try_delete(SANITY_FILE)`?  

This is how the test ensures that that sanity test will run each time.   I think its more explicit than the somewhat confusingly named `EM_IGNORE_SANITY`.   The intent of this test is to re-run the sanity checks each time around this loop.

Comment (sbc100) [reply tests/test_sanity.py:307]: As a followup I may propose removing `EM_IGNORE_SANITY` completely, but for now this just removes all the current users.

Comment (sbc100) [reply tests/test_sanity.py:307]: In the context of these tests a better name for `EM_IGNORE_SANITY` would be `EM_FORCE_SANITY`

Comment (kripken) [reply tests/test_sanity.py:307]: I see, thanks, so `EM_IGNORE_SANITY` was why that line was not needed before... yes, this is better I agree.

**Review by kripken — APPROVED** [review]

**Pull request merged**